idp/openid/openid-connect.go

```diff
@@ -397,16 +418,31 @@ func (idp *openidConnectIdentityProvider) CreateIdentity(ctx context.Context, to
 		}
 		user.Email = claims.Email
 		user.Name = claims.FullName
+		user.Groups, err = idp.RetrieveGroups(ctx, tok, &claims, id.Claims)
```

Comment:
`id.Claims(&claims)` was called above. Why does `id.Claims` then gets passed to `RetrieveGroups` for calling it again and checking the error?

Technically, and because of this, it would be impossible to reach that "Failed to unmarshal claims" error in `msgraph.go`.